src/main/java/uk/gov/ida/verifyserviceprovider/VerifyServiceProviderApplication.java

```diff
@@ -67,12 +65,14 @@ public String getName() {
     @Override
     public void run(VerifyServiceProviderConfiguration configuration, Environment environment) throws Exception {
         VerifyServiceProviderFactory factory = new VerifyServiceProviderFactory(configuration, environment);
+        ManifestReader manifestReader = new ManifestReader();
 
         environment.jersey().register(new JerseyViolationExceptionMapper());
         environment.jersey().register(new JsonProcessingExceptionMapper());
         environment.jersey().register(new InvalidEntityIdExceptionMapper());
-        environment.jersey().register(new VersionNumberResource(new ManifestReader()));
-        environment.jersey().register(factory.getGenerateAuthnRequestResource());
+        environment.jersey().register(new VersionNumberResource(manifestReader));
+
+        environment.jersey().register(factory.getGenerateAuthnRequestResource(manifestReader));
```

Comment:
Not really a problem, but to me the wiring up of the dependency of `VersionNumberResource` and `GenerateAuthnRequestResouce` on `ManifestReader` feels like something the factory should be doing, not the Application.

If `ManifestReader` is supposed to be a singleton then we can have a gross `synchronized` method in the factory, like we do with the metadata resolvers. Although looking at it it probably doesn't need to be.